spec/services/user_associations_service/create_spec.rb

```diff
@@ -3,97 +3,121 @@
 require 'rails_helper'
 
 RSpec.describe UserAssociationsService::Create, { can_edit_current_and_next_cycles: false } do
-  let(:user) { create(:user) }
+  let!(:user) { create(:user) }
+  let(:accredited_provider) { create(:provider, :accredited_provider, users: [user]) }
+  let!(:new_accredited_provider) { create(:provider, :accredited_provider, provider_code: 'AAA') }
+  let!(:next_new_accredited_provider) { create(:provider, :accredited_provider, :next_recruitment_cycle, users: [user], provider_code: 'AAA') }
```

Comment:
also not a blocker just a nit to mention `next_cycle_new_accredited_provider`

Reply:
Good shout